controllers/containerizedworkload_controller_helper.go

```diff
@@ -34,87 +29,58 @@ const (
 )
 
 // create a corresponding deployment
-func (r *ContainerizedWorkloadReconciler) renderWorkload(_ context.Context,
+func (r *ContainerizedWorkloadReconciler) renderDeployment(ctx context.Context,
 	workload *oamv1alpha2.ContainerizedWorkload) (*appsv1.Deployment, error) {
-	var RevisionHistoryLimit int32 = 100
-	deployName := workload.Name + "-deployment"
-	depl := appsv1.Deployment{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: appsv1.SchemeGroupVersion.String(),
-			Kind:       KindDeployment,
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      deployName,
-			Namespace: workload.Namespace,
-		},
-		Spec: appsv1.DeploymentSpec{
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{OAMResourceTypeLabel: string(workloadType), OAMResourceNameLabel: deployName},
-			},
-			RevisionHistoryLimit: &RevisionHistoryLimit,
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{OAMResourceTypeLabel: string(workloadType), OAMResourceNameLabel: deployName},
-				},
-				Spec: corev1.PodSpec{
-					Containers: workload.Spec.Containers,
-				},
-			},
-		},
+
+	resources, err := cwh.Translator(ctx, workload)
+	if err != nil {
+		return nil, err
+	}
+	deploy, ok := resources[0].(*appsv1.Deployment)
+	if !ok {
+		return nil, fmt.Errorf("internal error, deployment is not rendered correctly")
 	}
+	// TODO: fix bug in the lib
+	r.Log.Info("translate a deployment", "deploy", deploy.Spec.Template.Spec.Containers[0])
+	deploy.Namespace = workload.Namespace
```

Comment:
I was chatting with @negz about setting this and was originally thinking that we should mirror the namespace of the `ContainerizedWorkload` as you do here. However, in the remote case, we can't guarantee that the namespace will exist in the remote cluster. For that reason, we just opted for using `default` namespace for now, with the thought that we might set namespace via a `Scope` in the future. This looks like a valid solution for the local case for now 👍

Reply:
yeah, the K8s throws me a very helpful error message without it.